### ISSUE TITLE
RTC does not work.

### ISSUE THREAD

**Issue by xiaozhisky1** [huggingface/lerobot#2696]
### Ticket Type

🐛 Bug Report (Something isn't working)

### Environment & System Info

```Shell
- lerobot version: 0.4.3
- Platform: Linux-5.10.134-17.3.al8.x86_64-x86_64-with-glibc2.35
- Python version: 3.10.19
- Huggingface Hub version: 0.35.3
- Datasets version: 4.1.1
- Numpy version: 2.2.6
- PyTorch version: 2.7.1+cu126
- Is PyTorch built with CUDA support?: True
- Cuda version: 12.6
- GPU model: NVIDIA H20
- Using GPU in script?: <fill in>
```

### Description

I trained pi05 using my own dataset (where the actions are absolute joint angles). The final training loss reached 0.006.

Then I ran examples/rtc/eval_dataset.py.
`rtc=RTCConfig(enabled=True, prefix_attention_schedule=<RTCAttentionSchedule.EXP: 'EXP'>, max_guidance_weight=100.0, execution_horizon=8, debug=True, debug_maxlen=1000), device='cuda:0', output_dir='rtc_debug_output', seed=42, inference_delay=4, use_torch_compile=False, torch_compile_backend='inductor', torch_compile_mode='default', torch_compile_disable_cudagraphs=True)`

<img width="2397" height="1769" alt="Image" src="https://github.com/user-attachments/assets/284b28e6-7098-4614-938b-b15582a573ce" />

<img width="3260" height="1769" alt="Image" src="https://github.com/user-attachments/assets/c1f8794a-ffaa-4ded-b4ce-04195e4011ab" />

<img width="3314" height="1769" alt="Image" src="https://github.com/user-attachments/assets/1486bbff-777a-4405-90a3-73593cd9e02b" />

<img width="3261" height="1769" alt="Image" src="https://github.com/user-attachments/assets/c9f61848-3504-4908-be94-fa8005e75039" />

<img width="3319" height="1769" alt="Image" src="https://github.com/user-attachments/assets/4dfe6dda-1108-43f4-9cd5-0d294aeaff3b" />

However, when I run the same script with the following parameters:
`
python examples/rtc/eval_dataset.py \
    --policy.path=lerobot/pi05_libero_finetuned \
    --dataset.repo_id=HuggingFaceVLA/libero \
    --rtc.execution_horizon=8 \
    --device=cuda
`
everything works fine. I would like to know where the error might be occurring.

### Context & Reproduction

The training parameters are as follows:
`            "args": [
                "--dataset.repo_id=/mnt/model/wlz/real_stack_purple_toy_joint_lerobot",
                "--policy.type=pi05",
                "--output_dir=./outputs/pi05_training",
                "--job_name=pi05_training",
                "--policy.repo_id=wlz",
                "--policy.pretrained_path=lerobot/pi05_base",
                "--policy.compile_model=true",
                "--policy.gradient_checkpointing=true",
                "--wandb.enable=false",
                "--policy.dtype=bfloat16",
                "--steps=5000",
                "--policy.device=cuda",
                "--batch_size=32",
                "--policy.input_features={\"observation.images.image\":{\"shape\":[3,256,256],\"type\":\"VISUAL\"},\"observation.images.image2\":{\"shape\":[3,256,256],\"type\":\"VISUAL\"},\"observation.state\":{\"shape\":[7],\"type\":\"STATE\"}}",
                "--policy.output_features={\"action\":{\"shape\":[7],\"type\":\"ACTION\"}}"
            ]
`

### Relevant logs or stack trace

```Shell

```

### Checklist

- [ ] I have searched existing tickets to ensure this isn't a duplicate.
- [ ] I am using the latest version of the `main` branch.
- [ ] I have verified this is not an environment-specific problem.

**Comment by xiaozhisky1** [comment]
By the way, `num_inference_steps = 10`. In addition, I tried adjusting `max_guidance_weight = 100`, `execution_horizon = 32`, and `inference_delay = 16`, but the trajectory did not show any noticeable changes.

**Comment by xiaozhisky1** [comment]
Sorry, this was my mistake—I forgot to remove a piece of debugging code. RTC does work as expected.

**Comment by helper2424** [comment]
@xiaozhisky1 no problem - ping me if something won't work well
